driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -115,7 +115,7 @@ public void run() {
 
         int toCreate = coreSize;
 
-        if (reusedConnection != null && reusedConnection.setOwner(this)) {
+        if (reusedConnection != null && reusedConnection.setOwner(this) && toCreate > 0) {
```

Comment:
I think we should change this slighty to:

> if (reusedConnection != null && toCreate > 0 && reusedConnection.setOwner(this)) {

This way we don't invoke `setOwner` on the connection to this pool unless `toCreate` is greater than 0.  As we don't want to assign ownership unless we will actually do it because then the connection will not get closed in `Cluster.onUp` where we check if the connection has an owner and if not close it.